content/en/apps/tutorials/application-tests.md

```diff
@@ -0,0 +1,600 @@
+  before(async () => { return await harness.start(); });
+  after(async () => { return await harness.stop(); });
+```
+
+Any logic that you want to execute before and after each test can be placed in the `beforeEach()` and `afterEach()` hooks. Clear the state of the harness and set the date to a fixed value before each test run:
+```js
+  beforeEach(
+    async () => {
+      await harness.clear();
+      await harness.setNow('2000-01-01');
+    });
+```
+
+After each test run, assert that there are no errors in the console.
+```js
+  afterEach(() => {
+    expect(harness.consoleErrors).to.be.empty;
+  });
+```
+
+If you want to learn more about these hooks, refer to this [Mocha resource](https://mochajs.org/#hooks). Feel free to customize the hooks as you see fit.
+
+Let's look at a more detailed example. <a name="assessment-form-test">Here</a> is a test case for the Assessment form that was covered in the [previous tutorial]({{< ref "apps/tutorials/app-forms" >}}):
+
+{{< highlight js "linenos=table,hl_lines=26 31 36,anchorlinenos=true,lineanchors=assessment-form-test" >}}
+  it('unit test confirming assessment with cough since 7 days', async () => {
+    // Load the assessment form and fill 'yes' on the first page and '7' on the second page
+    const result = await harness.fillForm('assessment', ['yes'], ['7']);
+
+    // Verify that the form successfully got submitted
+    expect(result.errors).to.be.empty;
+
+    // Verify some attributes on the resulting report
+    expect(result.report.fields).to.deep.include({
+      patient_name: 'Patient Name',
+      group_assessment: {
+        cough: 'yes',
+        cough_duration: '7',
+      }
+    });
+  });
+{{< /highlight >}}
+
+In [line 3](#assessment-form-test-3) above, the method [harness.fillForm()](https://docs.communityhealthtoolkit.org/cht-conf-test-harness/Harness.html#fillForm) fills the specified form with the given input (answers).
+
+Depending on the form design, the number of inputs to be filled can be large. The inputs are often repeated within a single test or across multiple tests with little or no variation. It is a good idea to keep them in a separate file and refer them from the tests as required. You can also introduce some variations in the inputs using function parameters.
+
+Example: `form-inputs.js`
+```js {linenos=table}
+module.exports = {
+  assessments: {
+    no_cough: [
+      ['no']
+    ],
+    cough: (days) => [
+      ['yes'],
+      [ days ]
+    ]
+  }
+};
+```
+
+In the test below, the inputs are used from the `form-inputs.js` file above:
+```js
+const { assessments } = require('../form-inputs');
+...
+const result = await harness.fillForm('assessment', ...assessments.cough('3'));
+```
+
+The test files are usually grouped in folders to read and run them easily. One way of grouping them is by creating folders for each of the components that can be tested: forms, contact summary, tasks, and targets.
+
+![alt_text](tests-dir.png "Test directory structure")
```

Comment:
Please replace with {{< figure src="tests-dir.png" link="tests-dir.png" alt="" title="" >}}
Style guide recommends above format instead of ![Alt Text]

Reply:
I agree. Thanks!